Expo quickstart (#10134)
auth0/docs: articles/quickstart/native/react-native-expo/interactive.md

seo_alias: react-native
interactive: true
files:
  - files/app-json
  - files/app
github:
  path: 00-Login
topics: 
  - quickstarts 
  - native 
  - react-native
---

# Add Login to your React Native App

<!-- markdownlint-disable MD002 MD012 MD041 -->

::: warning
This SDK is not compatible with "Expo Go" app. It is compatible only with Custom Dev Client and EAS builds
:::

## Configure Auth0 {{{ data-action=configure }}}

To use Auth0 services, you need to have an application set up in the Auth0 Dashboard. The Auth0 application is where you will configure authentication in your project.

### Configure an application

Use the interactive selector to create a new Auth0 application or select an existing application that represents the project you want to integrate with. Every application in Auth0 is assigned an alphanumeric, unique client ID that your application code will use to call Auth0 APIs through the SDK.

Any settings you configure using this quickstart will automatically update for your Application in the <a href="${manage_url}/#/">Dashboard</a>, which is where you can manage your Applications in the future.

> If you would rather explore a complete configuration, you can view a sample application instead.

Reviewer: I would link "a sample application" to the GitHub repo.

Author: Looks like this sentence is something standard we have used in many places. And we can't link "Sample Application" as it refers to the one inside the dashboard AFAIK.

<img width="293" alt="Screenshot 2022-09-09 at 10 33 57" src="https://user-images.githubusercontent.com/15910425/189308148-1e7444a4-96ab-41ec-9a57-11c7b607dd3f.png">

- for Android - `{ANDROID_PACKAGE}://${account.namespace}/android/{ANDROID_PACKAGE}/callback`
:::

### Configure Logout URLs

A logout URL is the application URL Auth0 will redirect your users to once they log out. If you do not set this value, users will not be able to log out from your application and will receive an error.

::: note
If you are following along with our sample project, set this
- for iOS - `{IOS_BUNDLE_IDENTIFIER}://${account.namespace}/ios/{IOS_BUNDLE_IDENTIFIER}/callback`
- for Android - `{ANDROID_PACKAGE}://${account.namespace}/android/{ANDROID_PACKAGE}/callback`
:::

## Install dependencies 

In this section, you will learn how to install the React Native Auth0 module.

::: note
Please refer to the [official documentation](https://facebook.github.io/react-native/) for additional details on React Native.
:::

### Yarn

```bash
yarn add react-native-auth0
```

::: note
For further reference on yarn, check [their official documentation](https://yarnpkg.com/en/docs).
:::

### npm

```bash
npm install react-native-auth0 --save
```

## Setup Auth0 Config Plugin {{{ data-action=code data-code="app.json#10:15"}}}

> The Auth0 package runs custom native code that needs to be configured at build time. We will use [Expo Config Plugin](https://docs.expo.dev/guides/config-plugins/) to achieve this.

Reviewer: ```suggestion
The Auth0 package runs custom native code that must be configured at build time. Use [Expo Config Plugin](https://docs.expo.dev/guides/config-plugins/) to achieve this.
```

> The `react-native-auth0` plugin will be added in the [Expo Config](https://docs.expo.dev/workflow/configuration/)

Reviewer: ```suggestion
The `react-native-auth0` plugin will be added in the [Expo config](https://docs.expo.dev/workflow/configuration/)
```

> You can replace `{DOMAIN}` with the value from Auth0 Dashboard under your Application's Settings.

Reviewer: Not needed if the correct placeholder is being used in `app-json.md` (see my other comment)

## Generate Native Source Code {{{ data-action=code data-code="app.json#31:36"}}}

> We have to generate the native code for the above configuration to be set. To do this we will run the following command

Reviewer: ```suggestion
You must generate the native code for the above configuration to be set. To do this, run the following command:
```

```bash
expo prebuild
```

> While generating the native source code, if the [Android package](https://github.com/expo/fyi/blob/main/android-package.md) and [iOS bundle identifier](https://github.com/expo/fyi/blob/main/bundle-identifier.md) are not already present in Expo Config, you will be prompted to provide it.

Reviewer: Same comments as previous, I guess these are copied:

```suggestion
You will be prompted to provide the [Android package](https://github.com/expo/fyi/blob/main/android-package.md) and [iOS bundle identifier](https://github.com/expo/fyi/blob/main/bundle-identifier.md) if they are not already present in the Expo config:
```

```bash
? What would you like your Android package name to be? > com.auth0samples

? What would you like your iOS bundle identifier to be? > com.auth0samples
```

> These values are used to set the callback and logout URL

Reviewer: Be mindful of punctuation:

```suggestion
These values are used to set the callback and logout URLs.
```
